src/providers/mod.rs

```diff
@@ -34,11 +34,19 @@ pub mod vagrant_virtualbox;
 
 use std::collections::HashMap;
 use std::fs::{self, File};
+#[cfg(not(feature = "legacy"))]
+use std::io::ErrorKind::NotFound;
 use std::io::prelude::*;
 use std::path::Path;
 
+use openssh_keys::PublicKey;
+use users::{self, User};
+#[cfg(feature = "legacy")]
 use update_ssh_keys::{AuthorizedKeys, AuthorizedKeyEntry};
-use users;
+#[cfg(not(feature = "legacy"))]
+use users::os::unix::UserExt;
+#[cfg(not(feature = "legacy"))]
+use tempfile;
```

Comment:
I think you can plainly drop this.

Reply:
Discussed OOB: `tempfile::foo` is supported in newer compiler, but our baseline toolchain doesn't have that yet. Let's keep this.